src/EdgeConProblem/EdgeConReduction.c

```diff
@@ -178,138 +267,157 @@ static g_context_s* init_g_context(Z3_context z3_ctx, EdgeConGraph graph, int co
+                disjunctionTab[disjunctionTabId] = P_(j, j_prime);
+            }
         }
+        formulaTab[tabId++] = Z3_mk_or(ctx->z3_ctx, disjunctionTabId, (Z3_ast *) disjunctionTab);
     }
 
-    Z3_ast res1 = Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) resTab1);
-    
-    int tab2Size = ((ctx->C_H - 1) * (ctx->C_H - 1) * (ctx->C_H - 1)) * 2;
-    Z3_ast resTab2[tab2Size];
-    tabId = 0;
+    return Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) formulaTab);
+}
 
-    for(int j = 2; j < ctx->C_H; j++){
+static Z3_ast build_phi_3_2(const g_context_s *ctx) { 
+    int nb_variables = ((ctx->C_H - 1) * (ctx->C_H - 1) * (ctx->C_H - 1)) * 2;
+    Z3_ast forlumaTab[nb_variables];
+    int tabId = 0;
+
+    for(int j = 1; j < ctx->C_H; j++){
         for(int j_prime = j + 1; j_prime < ctx->C_H; j_prime++){
             for(int j_prime2 = j_prime + 1; j_prime2 < ctx->C_H; j_prime2++){
-                Z3_ast p_j_jprime = P_(j, j_prime);
-                Z3_ast p_j_jprime2 = P_(j, j_prime2);
-                Z3_ast not_p_j_jprime = NOT(p_j_jprime);
-                Z3_ast not_p_j_jprime2 = NOT(p_j_jprime2);
-                resTab2[tabId++] =  OR(2) 
-                                        not_p_j_jprime,
-                                        not_p_j_jprime2
-                                    EOR;
+                forlumaTab[tabId++] =  
+                    OR(2) 
+                        NOT( P_(j, j_prime) ),
+                        NOT( P_(j, j_prime2) )
+                    EOR;
             }
         }
     }
-    
-    Z3_ast res2 = Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) resTab2);
-    Z3_ast res[2] = {res1, res2}; 
-    return Z3_mk_and(ctx->z3_ctx, 2, res);
+
+    return Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) forlumaTab);
 }
 
+static Z3_ast build_phi_3(const g_context_s *ctx) { 
+    return (
+        AND(2)
+            build_phi_3_1(ctx),
+            build_phi_3_2(ctx)
+        EAND
+    );
+}
 
-static Z3_ast build_phi_4(const g_context_s *ctx){
-    int tab1Size = ctx->C_H * (ctx->N);
-    Z3_ast resTab1[tab1Size];
+static Z3_ast build_phi_4_1(const g_context_s *ctx){
+    Z3_ast formulaTab[ctx->C_H];
     int tabId = 0;
 
-    for(int i = 1; i < ctx->C_H; i++){
-        for(int n = 1; i < ctx->N; n++){
-            resTab1[tabId++] = L_(i, n);
+    for(int i = 0; i < ctx->C_H; i++){
+        Z3_ast disjunctionTab[ctx->N];
+        int disjunctionTabId = 0
+        for(int n = 0; i < ctx->N; n++){
+            disjunctionTab[disjunctionTabId] = L_(i, n);
         }
+        formulaTab[tabId++] = Z3_mk_or(ctx->z3_ctx, disjunctionTabId, (Z3_ast *) disjunctionTab);
     }
 
-    Z3_ast res1 = Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) resTab1);
+    return Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) formulaTab);
+}
 
-    int tab2Size = ctx->C_H * ((ctx->N * ctx->N) / 2) * 2;
-    Z3_ast resTab2[tab2Size];
+static Z3_ast build_phi_4_2(const g_context_s *ctx){
+    int nb_variables = ctx->C_H * ((ctx->N * ctx->N) / 2) * 2;
```

Comment:
**issue:** Why `... * 2`? And you don't need to use a auxiliary variable for storing the size of the array.